app/models/user.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  has_secure_password
+  validates :name, presence: true, length: {
+    minimum: 5
+  }
+  before_create :capitalize_name
+  validates :birthday, presence: true
+  validate :birthday_within_last_100_years
+
+  private
+
+  def capitalize_name
+    self.name = name.upcase
+  end
+
+  def birthday_within_last_100_years
+    return unless birthday.present? && birthday < 100.years.ago.to_date
+
+    errors.add(:birthday, "must be within the last 100 years")
```

Comment:
Đặt hết text vào i18n, thử bằng add errors với key lỗi
https://guides.rubyonrails.org/active_record_validations.html#:~:text=7.4-,errors.add,-The%20add%20method